### ISSUE TITLE
Naming/VariableNumber false positive for integer symbols

### ISSUE THREAD

**Issue by Pritilender** [rubocop-hq/rubocop#9041]
--------

## Expected behavior

The `Naming/VariableNumber` shouldn't report having integers as symbols (`:"1"` or `%i[1 2]`) as violations.

## Actual behavior

When Rubocop spots `:"1"` or `%i[1 2]` it reports it as `Naming/VariableNumber` violation.

```
tmp/rubocop_proof.rb:3:4: C: Naming/VariableNumber: Use normalcase for symbol numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)
%i[1 2 3.14 foo]
   ^
tmp/rubocop_proof.rb:3:6: C: Naming/VariableNumber: Use normalcase for symbol numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)
%i[1 2 3.14 foo]
     ^
tmp/rubocop_proof.rb:4:1: C: Naming/VariableNumber: Use normalcase for symbol numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)
:"1"
^^^^
```
## Steps to reproduce the problem

Create a Ruby file with following content:
```ruby
%i[1 2 3.14 foo]
:"1"
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
1.3.0 (using Parser 2.7.2.0, rubocop-ast 1.1.1, running on ruby 2.7.2 x86_64-darwin19)
  - rubocop-performance 1.8.1
  - rubocop-rails 2.8.1
  - rubocop-rspec 2.0.0
```
